HIRS_Utils/src/main/java/hirs/data/persist/certificate/Certificate.java

```diff
@@ -328,6 +331,47 @@ public Certificate(final byte[] certificateBytes) throws IOException {
         this.certAndTypeHash = Objects.hash(certificateHash, getClass().getSimpleName());
     }
 
+    @SuppressWarnings("magicnumber")
+    private byte[] trimCertificate(final byte[] certificateBytes) {
+        int certificateStart = 0;
+        int certificateLength = 0;
+
+        StringBuilder malformedCertStringBuilder = new StringBuilder(MALFORMED_CERT_MESSAGE);
+        for (int i = 0; i < certificateBytes.length; i++) {
+            if (i == certificateBytes.length - 1) {
```

Comment:
Throw an error if we reach the last byte of the provided "certificateBytes" since no Certificate length field could be found.